r/R/main.R

```diff
@@ -104,11 +103,27 @@ calc_Ks <- function(k_list, TC=25, S=35, Mg=0.0528171, Ca=0.0102821, P=NULL, MyA
     ks_list[[k]] <- calc_K(k = k, TC=TC,
                            S=S, Mg=Mg, 
                            Ca=Ca, P=P, 
-                           MyAMI_calc=MyAMI_calc)
+                           MyAMI_calc=MyAMI_calc,
+                           run_MyAMI=FALSE)
   }
     
   # Return data.frame
   Ks = data.frame(t(do.call(rbind, ks_list)))
+  
+  # Calculate correction factor with MyAMI
+  if(MyAMI_calc == TRUE){ 
+    Fcorr = pymyami$calc_Fcorr(Sal=S, TempC=TC, Mg=Mg, Ca=Ca)
+  } else {
+    Fcorr = pymyami$approximate_Fcorr(Sal=S, TempC=TC, Mg=Mg, Ca=Ca)
+  }
+    
+  # Apply correction
+  KF = Fcorr[[k]]
```

Comment:
At written, this block applies a single Fcorr to all the Ks, whereas the Fcorr for each K is different. This needs to loop through all the Ks and apply each Fcorr to each K separately.

This is causing crosscheck to fail.

Reply:
Yes, must have missed that. Corrected in new PR.